Hardware/Hardware/CardReader/UnderlyingError.swift

```diff
@@ -0,0 +1,308 @@
+        case .unsupportedReaderVersion:
+            return NSLocalizedString("The card reader software is out-of-date - please update the card reader software before attempting to process payments",
+                                     comment: "Error message when the card reader software is too far out of date to process payments.")
+        case .connectFailedReaderIsInUse:
+            return NSLocalizedString("Unable to connect to card reader - the card reader is already in use",
+                                     comment: "Error message when attempting to connect to a card reader which is already in use.")
+        case .unexpectedSDKError:
+            return NSLocalizedString("The system experienced an unexpected software error",
+                                     comment: "Error message when the card reader service experiences an unexpected software error.")
+        case .paymentDeclinedByPaymentProcessorAPI:
+            return NSLocalizedString("The card was declined by the payment processor - please try another means of payment",
+                                     comment: "Error message when the card processor declines the payment.")
+        case .paymentDeclinedByCardReader:
+            return NSLocalizedString("The card was declined by the card reader - please try another means of payment",
+                                     comment: "Error message when the card reader itself declines the card.")
+        case .notConnectedToInternet:
+            return NSLocalizedString("No connection to the Internet - please connect to the Internet and try again",
+                                     comment: "Error message when there is no connection to the Internet.")
+        case .requestTimedOut:
+            return NSLocalizedString("The request timed out - please try again",
+                                     comment: "Error message when a request times out.")
+        case .readerSessionExpired:
+            return NSLocalizedString("The card reader session has expired - please disconnect and reconnect the card reader and then try again",
+                                     comment: "Error message when the card reader session has timed out.")
+        case .processorAPIError:
+            return NSLocalizedString("The payment can not be processed by the payment processor.",
+                                     comment: "Error message when the payment can not be processed (i.e. order amount is below the minimum amount allowed.)")
+        case .internalServiceError:
+            return NSLocalizedString("Sorry, this payment couldn’t be processed",
+                                     comment: "Error message when the card reader service experiences an unexpected internal service error.")
+        case .incompleteStoreAddress(_):
+            return NSLocalizedString("The store address is incomplete or missing, please update it before continuing.",
```

Comment:
See comment above about invalid too. Might want to reword this if that is the case as "incomplete or invalid."